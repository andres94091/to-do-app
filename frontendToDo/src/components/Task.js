import React, { useState } from 'react'
import axios from 'axios'
// import PropTypes from 'prop-types'

const Task = ({ todo, index, completeTodos, setBandReload }) => {

    const [task, setTask] = useState(todo)


    async function updateTask() {
        try {
            await axios.patch(`http://localhost:5000/api/tasks/${task.id}`, {
                status: !task.status
            })
            setBandReload(band => !band)
        } catch (e) {
            console.log(e);
        }
    }

    function handleClick() {
        completeTodos(index)
        setTask(({ id, status, task, user_id }) => {
            return {
                id,
                status: !status,
                task,
                user_id
            }
        })
        updateTask()
    }

    return (
        <div
            className='todo'
            style={{ textDecoration: todo.status ? "line-through" : "" }}
        >
            {todo.task}
            <div>
                <button onClick={() => handleClick()}> status</button>
            </div>
        </div>
    )
}

// task.propTypes = {

// }

export default Task
