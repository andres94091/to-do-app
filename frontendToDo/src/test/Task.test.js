import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from 'enzyme';

import Task from '../components/task';

describe('tests in component Task', () => {
    
    test('should render the component correctly', () => {

        const todo = {task: 'test task', status: false, user_id: 1}
        const wrapper = shallow(<Task todo={todo} index ={0} />)

        expect(wrapper).toMatchSnapshot();
        
    })
})
