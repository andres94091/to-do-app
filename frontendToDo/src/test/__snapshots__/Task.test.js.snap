// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in component Task should render the component correctly 1`] = `
<div
  className="todo"
  style={
    Object {
      "textDecoration": "",
    }
  }
>
  test task
  <div>
    <button
      onClick={[Function]}
    >
       status
    </button>
  </div>
</div>
`;
